source/patterns/@aws-solutions-constructs/core/lib/elasticsearch-helper.ts

```diff
@@ -217,3 +217,115 @@ export function buildElasticSearchCWAlarms(scope: Construct): cloudwatch.Alarm[]
 
   return alarms;
 }
+
+function retrievePrivateSubnetIds(vpc: ec2.IVpc) {
+  let targetSubnetType;
+
+  if (vpc.isolatedSubnets.length) {
+    targetSubnetType = ec2.SubnetType.PRIVATE_ISOLATED;
+  } else if (vpc.privateSubnets.length) {
+    targetSubnetType = ec2.SubnetType.PRIVATE_WITH_NAT;
+  } else {
+    throw new Error('Error - ElasticSearch Domains can only be deployed in Isolated or Private subnets');
+  }
+
+  const subnetSelector = {
+    onePerAz: true,
+    subnetType: targetSubnetType
+  };
+
+  return vpc.selectSubnets(subnetSelector).subnetIds;
+}
+
+function createClusterConfiguration(zoneLength?: number): elasticsearch.CfnDomainProps {
```

Comment:
My implementation was much shorter

```
function CreateClusterConfiguration(
  numberOfAzs?: number
): elasticsearch.CfnDomain.ElasticsearchClusterConfigProperty {

  return {
    zoneAwarenessEnabled: true,
    zoneAwarenessConfig: {
      availabilityZoneCount: numberOfAzs,
    },
    instanceCount: numberOfAzs,
  };
}
```

I see one reason this got longer which we can roll back, what other reasons? zoneLength doesn't seem like as good a variable name as numberOf Azs. 

I seem to recall you saying that even when there are 2 AZs there is a setting that needs to be 3. Can we add dedicatedMasterCount: 3 to my version and cover that possibliity?

Reply:
Yes, I added dedicatedMasterCount and dedicatedMasterEnabled to your configuration